pkgs/development/compilers/ghc/9.0.2.nix

```
@@ -106,8 +106,8 @@ let
   '' + lib.optionalString (!enableProfiledLibs) ''
     GhcLibWays = "v dyn"
   '' + lib.optionalString enableRelocatedStaticLibs ''
-    GhcLibHcOpts += -fPIC
-    GhcRtsHcOpts += -fPIC
+    GhcLibHcOpts += -fPIC -fexternal-dynamic-refs
+    GhcRtsHcOpts += -fPIC -fexternal-dynamic-refs
```

Sorry, GitHub isn't letting me create an actual suggestion here, but how about writing this as:

```suggestion
  '' + 
  # These options are recommended by
  # https://www.tweag.io/blog/2020-09-30-bazel-static-haskell/
  # We're not sure why -fexternal-dynamic-refs is necessary.
  lib.optionalString enableRelocatedStaticLibs ''
    GhcLibHcOpts += -fPIC -fexternal-dynamic-refs
    GhcRtsHcOpts += -fPIC -fexternal-dynamic-refs
```

This same change should be made for `9.2.1.nix` as well.

---

Thanks, I added something like this.